pumpkin-lib/src/engine/sat/assignments_propositional.rs

```diff
@@ -221,12 +223,15 @@ impl AssignmentsPropositional {
         self.assignment_info[variable] = PropositionalAssignmentInfo::Unassigned;
     }
 
+    /// Enqueues a [`Literal`] which was the result of a decision by a [`Brancher`].
```

Comment:
This does not necessarily have to come from a rancher. When using the assignment as a standalone module, decisions can come from anywhere. I would suggest something like: 'Enqueues a literal as a decision. Unlike self::enqueue_propagated_literal, this enqueue has no reason.'